app/controllers/vm_infra_controller.rb

```diff
@@ -74,15 +74,45 @@ def skip_breadcrumb?
   menu_section :inf
   has_custom_buttons
 
+  def simple_dialog_initialize(ra, options)
+    @record = Dialog.find_by(:id => ra.dialog_id.to_i)
+    new = options[:dialog] || {}
+    @edit = {
+      :new             => new,
+      :wf              => ResourceActionWorkflow.new(new, current_user, ra, {}),
+      :rec_id          => @record.id,
```

Comment:
if @record is nil? is that ok as a rec_id? (asking only because in the next line you check for that condition)

for the record.. this is where tests are really useful....

Reply:
I check whether `@record.id` is nil, not `@record`. I think, this is OK for `rec_id`, because the same is happening here: https://github.com/ManageIQ/manageiq-ui-classic/blob/cdc7af2557dc20d9f6b2ffa4e67392463eb00d64/app/controllers/application_controller/dialog_runner.rb#L147

Reply:
I disagree, (maybe there is a bug in another place too) but if @record is nil you'll get:
```
NoMethodError: undefined method `id' for nil:NilClass
```

Reply:
@ohadlevy Done.